### PR TITLE
Add CeoCard as a support class like the other card types.

### PR DESCRIPTION
**Pull request by kberg**
This follows on from #5297. It reduces repetition, and also uses the StaticCard pattern which significantly reduces memory consumption.

### CODE REVIEW

**Comment by d-little** [comment]
Amazing work



Any reason for the difference in approach for Naomi and Yvonne canAct overrides?  is the canAct() better, performance wise?  Or negligible.  They're in opposite evaluating order


Naomi:
```ts
  public override canAct(player: Player): boolean {
    if (!super.canAct(player)) {
      return false;
    }
    return openColonies.length > 0;
```

Yvonne :
```ts
  public override canAct(player: Player): boolean {
    return player.game.gameOptions.coloniesExtension === true && super.canAct(player);
  }
```

Should they be:

```ts
  public override canAct(player: Player): boolean {
    if (!super.canAct(player)) return false;
    return openColonies.length > 0;
```

```ts
  public override canAct(player: Player): boolean {
    if (!super.canAct(player)) return false;
    return player.game.gameOptions.coloniesExtension;
```

File: src/server/cards/ceos/Naomi.ts [hunk 1]
```diff
@@ -29,15 +25,12 @@ export class Naomi extends Card implements ICeoCard {
     });
   }
 
-  public isDisabled = false;
-
-  public override play() {
-    return undefined;
-  }
-
-  public canAct(player: Player): boolean {
```
Comment (d-little) [inline src/server/cards/ceos/Naomi.ts:38]: Any reason for the difference in approach for Naomi and Yvonne canAct overrides?  is the canAct() better, performance wise?  Or negligible.  They're in opposite evaluating order


Naomi:
```ts
  public override canAct(player: Player): boolean {
    if (!super.canAct(player)) {
      return false;
    }
    return openColonies.length > 0;
```

Yvonne :
```ts
  public override canAct(player: Player): boolean {
    return player.game.gameOptions.coloniesExtension === true && super.canAct(player);
  }
```

Should they be:

```ts
  public override canAct(player: Player): boolean {
    if (!super.canAct(player)) return false;
    return openColonies.length > 0;
```

```ts
  public override canAct(player: Player): boolean {
    if (!super.canAct(player)) return false;
    return player.game.gameOptions.coloniesExtension;
```

Comment (kberg) [reply src/server/cards/ceos/Naomi.ts:38]: It's a little arbitrary, but I'm using these rules.

1. Do faster evaluations first if possible.
2. Make it readable.

Comment (kberg) [reply src/server/cards/ceos/Naomi.ts:38]: Rules/Guidelines

**Review by d-little — APPROVED** [review]

**Comment by d-little** [comment]
Looks good, assuming these tests pass, send it

**Pull request merged**